src/CoreShop/Bundle/CoreBundle/Report/ManufacturerReport.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * CoreShop.
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2015-2017 Dominik Pfaffenbauer (https://www.pfaffenbauer.at)
+ * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
+ */
+
+namespace CoreShop\Bundle\CoreBundle\Report;
+
+use Carbon\Carbon;
+use CoreShop\Component\Core\Model\ProductInterface;
+use CoreShop\Component\Core\Model\StoreInterface;
+use CoreShop\Component\Core\Report\ReportInterface;
+use CoreShop\Component\Currency\Formatter\MoneyFormatterInterface;
+use CoreShop\Component\Locale\Context\LocaleContextInterface;
+use CoreShop\Component\Order\OrderStates;
+use CoreShop\Component\Pimcore\DataObject\InheritanceHelper;
+use CoreShop\Component\Product\Model\ManufacturerInterface;
+use CoreShop\Component\Resource\Repository\PimcoreRepositoryInterface;
+use CoreShop\Component\Resource\Repository\RepositoryInterface;
+use Doctrine\DBAL\Connection;
+use Pimcore\Model\DataObject;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+class ManufacturerReport implements ReportInterface
+{
+    /**
+     * @var int
+     */
+    private $totalRecords = 0;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $storeRepository;
+
+    /**
+     * @var Connection
+     */
+    private $db;
+
+    /**
+     * @var MoneyFormatterInterface
+     */
+    private $moneyFormatter;
+
+    /**
+     * @var LocaleContextInterface
+     */
+    private $localeService;
+
+    /**
+     * @var PimcoreRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var PimcoreRepositoryInterface
+     */
+    private $orderItemRepository;
+
+    /**
+     * ManufacturerReport constructor.
```

Comment:
no need for this description

Reply:
https://github.com/coreshop/CoreShop/search?l=PHP&q=%22+constructor.%22
Then we should remove them globally.

Reply:
true